Add explicit support for React 17

React 17 has been released in October (see [blog post](https://reactjs.org/blog/2020/10/20/react-v17.html)) but this package accepts only `react@^16.3.0` as a peer dependency, so a warning is displayed when installing:

> react-share@4.3.1 requires a peer of react@^16.3.0 but none is installed. You must install peer dependencies yourself.

For my use cases, the package seems to work fine with v17 but I'm not sure if it is safe to just add v17 as a possible dependency without further evaluation.

Note that this is more problematic w/ npm 7, which tries to resolve peer deps. So we get errors like:

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: eyas-blog@0.1.0
npm ERR! Found: react@17.0.1
npm ERR! node_modules/react
npm ERR!   react@"^17.0.1" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.3.0" from react-share@4.3.1
npm ERR! node_modules/react-share
npm ERR!   react-share@"^4.3.1" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

--- a/package.json
+++ b/package.json
@@ -91,6 +91,6 @@
     "jsonp": "^0.2.1"
   },
   "peerDependencies": {
-    "react": "^16.3.0"
+    "react": ">=16.3.0"
   }
 }

